### PR TITLE
Remove deprecated SSL settings and simplify SSL configuration

### PR DESCRIPTION
**Pull request by donoghuc**
Get rid of the deprecated parameters and document their removal. Ensure the integration and unit tests pass and that the removals are clearly documented. 

Closes https://github.com/logstash-plugins/logstash-input-elasticsearch/issues/210

### CODE REVIEW

**Comment by donoghuc** [comment]
Got a failure on 7.x https://app.travis-ci.com/github/logstash-plugins/logstash-input-elasticsearch/jobs/628037127 that looked unrelated and suspiciously like a race condition. I reloaded that cell.

File: lib/logstash/inputs/elasticsearch.rb
Comment (donoghuc) [inline lib/logstash/inputs/elasticsearch.rb:396]: oops, i should only set this if `@ssl_enabled`. I'll move that back to where it was before.

**Review by robbavey — CHANGES_REQUESTED** [review]
Partial review while we finalize the exact wording in docs, but let's get started with marking old settings as `obsolete` before removal

File: lib/logstash/inputs/elasticsearch.rb
Comment (donoghuc) [reply lib/logstash/inputs/elasticsearch.rb:201]: ah, I totally missed that. Will get this fixed up!

**Comment by donoghuc** [comment]
TODO:

- [ ] Review https://github.com/logstash-plugins/logstash-input-http/pull/182/files and make sure we have consistency where possible
- [ ] Grep through the repo for the old settings (including docs, tests and comments) to find references to remove/update

File: spec/inputs/elasticsearch_spec.rb
Comment (donoghuc) [inline spec/inputs/elasticsearch_spec.rb:55]: Inspired by tests in https://github.com/logstash-plugins/logstash-input-http/pull/182 credit @robbavey

File: lib/logstash/inputs/elasticsearch.rb
Comment (robbavey) [inline lib/logstash/inputs/elasticsearch.rb:257]: Thinking about this, we can probably skip the `:validate` and `:default` parts, and stick with `:obsolete` to avoid users fixing up settings, only to find they were obsolete anyway. WDYT?

(My only caveat is to keep `:password` and `:uri` (not applicable here), as they are considered secret, and are [obfuscated](https://github.com/elastic/logstash/blob/a94659cf82d7e62c5a5f550df3e4e5bd0c6cf3e1/logstash-core/lib/logstash/config/mixin.rb#L608-L615) on discovery)

Comment (donoghuc) [reply lib/logstash/inputs/elasticsearch.rb:257]: Nice, yeah i got tripped up on this writing tests. I was surprised to find that `:validate` would trigger when the option is `:obsolete`. I may be misunderstanding the test scenario but it looks to me like as a user I may be frustrated with a case whereby i set the wrong thing, then get a validation error that its the wrong form, only to be rewarded with a "this is no longer a valid option" once I satisfy the input validation. Not a huge deal but certainly a surprise if that is the case.

File: docs/index.asciidoc
Comment (robbavey) [inline docs/index.asciidoc:613]: For the docs update, let's follow the example in https://github.com/logstash-plugins/logstash-output-http/commit/2fae93e26b103cf9e7c0ec0d257be903070805b9#diff-cae5619b3d18ec99c5ccd0a9f6de0c6d3f53343c64692444551a7d29da6863e7

ie, remove all inline references to deprecated settings, and add a dedicated section at the bottom, above "common options", with a link to that section just before the config table.

cc @karenzone

**Comment by donoghuc** [comment]
Rebased and updated documentation to match https://github.com/logstash-plugins/logstash-output-http/pull/147

**Comment by donoghuc** [comment]
Corresponding docs PR https://github.com/elastic/logstash/pull/16744 in LS

**Review by robbavey — CHANGES_REQUESTED** [review]
Copy/paste error and a comment nit

File: docs/index.asciidoc
Comment (donoghuc) [reply docs/index.asciidoc]: Yep, fixed in amended commit. Good catch.

**Comment by robbavey** [comment]
Should be good to go after a rebase

**Comment by donoghuc** [comment]
Pushed the rebase. I think outstanding work after merge is:

[] Publish 5.0.0
[] PIn to 4 series in LS 8.x [example](https://github.com/elastic/logstash/commit/1d82e377ffcd63c7fad855d1f0cb6ab77398dfde)
[] Ensure [changes are documented](https://github.com/elastic/logstash/pull/16744) in LS

**Review by robbavey — APPROVED** [review]
LGTM - over to you @karenzone to check out doc changes

File: docs/index.asciidoc
Comment (karenzone) [inline docs/index.asciidoc:99]:
```suggestion
This plugin supports these configuration options plus the <<plugins-{type}s-{plugin}-common-options>>  described later.
```

Comment (karenzone) [reply docs/index.asciidoc:99]: This suggestion:
* Removes antiquated language ("the following..." )
* Removes inaccurate info suggesting that we still support deprecated/obsolete settings
* Removes invalid reference so that we won't break the docs builds

**Review by karenzone — CHANGES_REQUESTED** [review]
Thanks for your work on this, @donoghuc. I left suggestion inline.

**Comment by donoghuc** [comment]
Thanks @karenzone I committed your suggestion.

**Comment by karenzone** [comment]
@donoghuc, one more thing I noticed after I got the docs building successfully. 
Looks like paragraph starting at line 633 was orphaned with removals.  

> Option to validate the server's certificate. Disabling this severely compromises security.
When certificate validation is disabled, this plugin implicitly trusts the machine
resolved at the given address without validating its proof-of-identity.
In this scenario, the plugin can transmit credentials to or process data from an untrustworthy
man-in-the-middle or other compromised infrastructure.
More information on the importance of certificate verification:
**https://www.cs.utexas.edu/~shmat/shmat_ccs12.pdf**.

Should it be deleted as part of `ssl_certificate_verification`?

**Comment by donoghuc** [comment]
You are right! I just deleted that and squashed all the commits to clean up before merging :)

File: docs/index.asciidoc
Comment (karenzone) [inline docs/index.asciidoc:105]:
```suggestion
```

Comment (karenzone) [reply docs/index.asciidoc:105]: Duplicate note

**Review by karenzone — APPROVED** [review]
Please delete duplicate note. Otherwise, LGTM!

File: CHANGELOG.md
Comment (donoghuc) [inline CHANGELOG.md:1]: Need to make sure this is not deleted.

**Pull request merged**
